### PR TITLE
Update kubernetes vendor

### LINKED ISSUE #66: Create pod failed: No command specified
Encountered when verifying PR #62 by running `kubectl run --image=nginx nginx`:

```
I0118 11:38:52.943145   23702 manager.go:177] CreateContainer with request pod_sandbox_id:"k8s_POD.0_nginx-2932712833-gc82w_default_78e43fe8-dd72-11e6-b3f3-fa163ea7515e_a76f41fe" config:<metadata:<name:"nginx" attempt:0 > image:<image:"" > working_dir:"" envs:<key:"KUBERNETES_SERVICE_HOST" value:"10.0.0.1" > envs:<key:"KUBERNETES_SERVICE_PORT" value:"443" >....
E0118 11:38:52.943961   23702 container.go:45] Create container nginx in pod k8s_POD.0_nginx-2932712833-gc82w_default_78e43fe8-dd72-11e6-b3f3-fa163ea7515e_a76f41fe failed: rpc error: code = 2 desc = No command specified
```

The problem is `image:<image:"" >`, which expected to be imageRef of nginx image.

cc/ @resouer @YaoZengzeng

### PR DESCRIPTION
**Pull request by feiskyer**
This PR updates kubernetes vendor and also fixes #66 #56.

### CODE REVIEW

File: pkg/hyper/images.go [hunk 1]
```diff
@@ -59,16 +59,22 @@ func (h *Runtime) ListImages(filter *kubeapi.ImageFilter) ([]*kubeapi.Image, err
 }
 
 // PullImage pulls the image with authentication config.
-func (h *Runtime) PullImage(image *kubeapi.ImageSpec, authConfig *kubeapi.AuthConfig) error {
```
Comment (resouer) [inline pkg/hyper/images.go]: e2e tests also need update (fix CI)

Comment (feiskyer) [inline pkg/hyper/images.go]: updated.

File: test/e2e/streaming.go [hunk 2]
```diff
@@ -128,7 +128,7 @@ func startLongRunningContainer(rc internalapi.RuntimeService, ic internalapi.Ima
 	imageSpec := &kubeapi.ImageSpec{
 		Image: &latestTestImageRef,
 	}
-	err = ic.PullImage(imageSpec, nil)
+	_, err = ic.PullImage(imageSpec, nil)
 	framework.ExpectNoError(err, "Failed to pull image: %v", err)
```
Comment (resouer) [inline test/e2e/streaming.go]: I'm afraid the streaming url has been changed on upstream: https://github.com/feiskyer/frakti/blob/60f69a711af0c98f378fe46a7fc535dd27c79fbd/test/e2e/streaming.go#L157

Comment (resouer) [inline test/e2e/streaming.go]: See: https://github.com/kubernetes/kubernetes/pull/38742/files#diff-399a1489053b508436d8c0845b122e4eR214

File: test/e2e/streaming.go [hunk 3]
```diff
@@ -128,7 +128,7 @@ func startLongRunningContainer(rc internalapi.RuntimeService, ic internalapi.Ima
 	imageSpec := &kubeapi.ImageSpec{
 		Image: &latestTestImageRef,
 	}
-	err = ic.PullImage(imageSpec, nil)
```
Comment (resouer) [inline test/e2e/streaming.go]: I'm afraid the streaming url has been changed on upstream: https://github.com/feiskyer/frakti/blob/60f69a711af0c98f378fe46a7fc535dd27c79fbd/test/e2e/streaming.go#L157

See: https://github.com/kubernetes/kubernetes/pull/38742/files#diff-399a1489053b508436d8c0845b122e4eR214

**Comment by feiskyer** [comment]
@resouer Fixed. Found another problem of filtering digest from hyperd. Dropping filters for a workaround.

**Comment by resouer** [comment]
lgtm

**Pull request merged**